src/elbo.jl

```diff
@@ -35,62 +36,56 @@ exact solution. If there is no such solution, `Default()` either uses
 `Quadrature()` or `MonteCarlo()`, depending on the likelihood.
 
 N.B. the likelihood is assumed to be Gaussian with observation noise `fx.Σy`.
-Further, `fx.Σy` must be homoscedastic and uncorrelated - i.e. `fx.Σy = α * I`.
+Further, `fx.Σy` must be spherical - i.e. `fx.Σy = α * I`.
 
 [1] - Hensman, James, Alexander Matthews, and Zoubin Ghahramani. "Scalable
 variational Gaussian process classification." Artificial Intelligence and
 Statistics. PMLR, 2015.
 """
 function AbstractGPs.elbo(
+    svgp::SVGP,
     fx::FiniteGP{<:AbstractGP,<:AbstractVector,<:Diagonal{<:Real,<:Fill}},
-    y::AbstractVector{<:Real},
-    fz::FiniteGP,
-    q::AbstractMvNormal;
+    y::AbstractVector{<:Real};
     n_data=length(y),
     method=Default(),
 )
-    return _elbo(method, fx, y, fz, q, GaussianLikelihood(fx.Σy[1]), n_data)
+    @assert svgp.fz.f === fx.f
+    return _elbo(method, svgp, fx, y, GaussianLikelihood(fx.Σy[1]), n_data)
 end
 
-function AbstractGPs.elbo(
-    ::FiniteGP, ::AbstractVector, ::FiniteGP, ::AbstractMvNormal; kwargs...
-)
+function AbstractGPs.elbo(::SVGP, ::FiniteGP, ::AbstractVector; kwargs...)
     return error(
         "The observation noise fx.Σy must be homoscedastic.\n To avoid this error, construct fx using: f = GP(kernel); fx = f(x, σ²)",
     )
 end
 
 """
-    elbo(lfx::LatentFiniteGP, y::AbstractVector, fz::FiniteGP, q::AbstractMvNormal; n_data=length(y), method=Default())
+    elbo(svgp, ::SVGP, lfx::LatentFiniteGP, y::AbstractVector; n_data=length(y), method=Default())
 
 Compute the ELBO for a LatentGP with a possibly non-conjugate likelihood.
 """
 function AbstractGPs.elbo(
-    lfx::LatentFiniteGP,
-    y::AbstractVector,
-    fz::FiniteGP,
-    q::AbstractMvNormal;
-    n_data=length(y),
-    method=Default(),
+    svgp::SVGP, lfx::LatentFiniteGP, y::AbstractVector; n_data=length(y), method=Default()
```

Comment:
```suggestion
    svgp::SVGP, lfx::LatentFiniteGP, y::AbstractVector; num_data=length(y), method=Default()
```
as mentioned in #32?